cluster-autoscaler/metrics/metrics.go

```diff
@@ -543,6 +562,24 @@ func UpdateNodeGroupTargetSize(targetSizes map[string]int) {
 	}
 }
 
+// UpdateNodeGroupHealthStatus records if node group is healthy to autoscaling
+func UpdateNodeGroupHealthStatus(nodeGroup string, healthy bool) {
+	if healthy {
+		nodesGroupHealthiness.WithLabelValues(nodeGroup).Set(1)
+	} else {
+		nodesGroupHealthiness.WithLabelValues(nodeGroup).Set(0)
+	}
+}
+
+// UpdateNodeGroupBackOffStatus records if node group is backoff for not autoscaling
+func UpdateNodeGroupBackOffStatus(nodeGroup string, backOff bool, errorClass cloudprovider.InstanceErrorClass) {
+	if backOff {
+		nodeGroupBackOffStatus.WithLabelValues(nodeGroup).Set(float64(errorClass))
```

Comment:
That is quite hacky and also meaningless when aggregated. Adding error codes together doesn't make sense and yet it is exactly what will happen when querying for multiple node groups. The previous approach with label for error code makes sense, you just need to make sure it stops reporting 1 when the node group is no longer in backoff.

Reply:
Thank you for your suggestion. I just wanted to confirm the expected metric behavior:
There are two ways to display this metrics since it involves different handling scenarios based on the reason. 
I am not sure which approach is more appropriate, and I would appreciate any advice you can provide.

Reply:
I think both are fine, though I slightly prefer the former. The reason is that whatever is collecting these metrics, will be able to clearly say that a given metric stream is "done" when backoff reason changes or goes away. With the second approach I think there can be artifacts on the graph, since a dashboard won't correlate different streams without some careful query crafting.

The way I would implement this is by keeping a map with {node_group, reason} pair as a key. It would then be updated in bulk:
1. Iterate through the map and set all values to 0
2. Iterate through the node groups and set corresponding {node_group, reason} value to 1 (or `true`). Optionally, inject {node_group, ""} keys set to 0 for all node groups that are not in backoff. Not sure if that second part is actually useful for anything, perhaps not.
3. Use a custom collector that will read values from the map when prometheus endpoint is scraped.

Reply:
Understood, thank you! and submit the code changes as soon as possible.

Reply:
Modifications done and PR submitted. Because the errorCode is provided by the nodeProvider and is open, this collector is dynamic and will search all historical collections of reported reasons.

Reply:
@x13n